packages/compass-e2e-tests/tests/collection-import.test.ts

```diff
@@ -990,6 +990,32 @@ describe('Collection import', function () {
     });
 
     it('aborts when disconnected', async function () {
+      process.env.COMPASS_E2E_TEST_IMPORT_ABORT_TIMEOUT = 'true';
+
+      // 16116 documents.
+      const csvPath = path.resolve(__dirname, '..', 'fixtures', 'listings.csv');
+
+      await browser.navigateToCollectionTab(
+        'test',
+        'import-abort',
+        'Documents'
+      );
+
+      // Open the import modal.
+      await browser.clickVisible(Selectors.AddDataButton);
+      const insertDocumentOption = await browser.$(Selectors.ImportFileOption);
+      await insertDocumentOption.waitForDisplayed();
+      await browser.clickVisible(Selectors.ImportFileOption);
+
+      // Select the file.
+      await browser.selectFile(Selectors.ImportFileInput, csvPath);
+
+      // Wait for the modal to appear.
+      const importModal = await browser.$(Selectors.ImportModal);
+      await importModal.waitForDisplayed();
+
+      // Confirm import.
+      await browser.clickVisible(Selectors.ImportConfirm);
```

Comment:
The diff doesn't make it obvious, but much of this test's logic was in the beforeEach(). I don't particularly like that style because it is test code and if it fails it is the beforeEach() that fails, not  the test. Also in this particular case it meant that there was no screenshot showing the failure (once ever the Import confirm button couldn't be clicked because it was disabled) because the screenshot is in an afterEach and the test never ran, so no screenshot.

As a general rule I try not to use the pattern of executing the test in hooks and only doing assertions in the test. Makes all the test code that can fail contained inside the test while only leaving genuine common code that has nothing to do with this specific test case in hooks. Easier to reason about as well and to isolate the effects from other tests in the file.